### PR TITLE
add new API field CommonTemplatesNamespace

### PR DESCRIPTION
**Pull request by ksimon1**
add new API field CommonTemplatesNamespace
This new field defines in which namespace will be templates deployed

/cc @tiraboschi @nunnatsa can you please help me, if these are all the places that need to be updated?

Signed-off-by: Karel Šimon <ksimon@redhat.com>

**Reviewer Checklist**
> Reviewers are supposed to review the PR for every aspect below one by one. To check an item means the PR is either "OK" or "Not Applicable" in terms of that item. All items are supposed to be checked before merging a PR. 

- [ ] PR Message
- [ ] Commit Messages
- [ ] How to test
- [ ] Unit Tests
- [ ] Functional Tests
- [ ] User Documentation
- [ ] Developer Documentation
- [ ] Upgrade Scenario
- [ ] Uninstallation Scenario
- [ ] Backward Compatibility
- [ ] Troubleshooting Friendly
  
**Release note**:

```
API Change: add commonTemplatesNamespace field to the HyperConverged's spec
```

### CODE REVIEW

File: pkg/controller/operands/ssp.go [hunk 1]
```diff
@@ -168,13 +168,18 @@ func (h *sspHooks) updateCr(req *common.HcoRequest, client client.Client, exists
 
 func NewSSP(hc *hcov1beta1.HyperConverged, opts ...string) *sspv1beta1.SSP {
```
Comment (nunnatsa) [inline pkg/controller/operands/ssp.go:169]: This is a good test, but please also add test/test case in pkg/controller/hyperconverged/hyperconverged_controller_test.go

**Review by nunnatsa — CHANGES_REQUESTED** [review]
Thanks for the PR, it looks good in general.

A few comments:
1. Please extend the PR comment to explain what is this new field and what is the reason to add it.
2. Please run `openapi-gen` and `deepcopy-gen`
3. Please document the new field in docs/cluster-configuration.md

File: pkg/controller/operands/ssp_test.go [hunk 2]
```diff
@@ -79,13 +79,14 @@ var _ = Describe("SSP Operands", func() {
...
 			expectedResource := NewSSP(hco)
 			existingResource := expectedResource.DeepCopy()
 			existingResource.ObjectMeta.SelfLink = fmt.Sprintf("/apis/v1/namespaces/%s/dummies/%s", existingResource.Namespace, existingResource.Name)
 
 			replicas := int32(defaultTemplateValidatorReplicas * 2) // non-default value
 			existingResource.Spec.TemplateValidator.Replicas = &replicas
-			existingResource.Spec.CommonTemplates.Namespace = "foobar"
 			existingResource.Spec.NodeLabeller.Placement = &lifecycleapi.NodePlacement{
 				NodeSelector: map[string]string{"foo": "bar"},
 			}
```
Comment (nunnatsa) [inline pkg/controller/operands/ssp_test.go:92]: This is a good test, but please also add test/test case in pkg/controller/hyperconverged/hyperconverged_controller_test.go

File: pkg/controller/hyperconverged/hyperconverged_controller_test.go [hunk 3]
```diff
@@ -279,6 +283,34 @@ var _ = Describe("HyperconvergedController", func() {
...
+				Eventually(func() bool {
+					err := cl.List(context.TODO(),
+						templatesList,
+						client.InNamespace(namespace),
+						client.MatchingLabels{
+							templateTypeLabel: base,
+						})
+					return len(templatesList.Items) > 0 && err == nil
+				}, 30*time.Second).Should(BeTrue(), "templates should be deployed in different namespace")
+			})
```
Comment (nunnatsa) [inline pkg/controller/hyperconverged/hyperconverged_controller_test.go:313]: This test is a unit test, without a running cluster. There is no running SSP to implement the logic under test. Also, as this is a unit test, it should not test other components. Please just make sure the required fields are set in the SSP CR.

File: pkg/controller/hyperconverged/hyperconverged_controller_test.go [hunk 4]
```diff
@@ -494,9 +526,9 @@ var _ = Describe("HyperconvergedController", func() {
 					Expect(requeue).To(BeFalse())
 					checkAvailability(foundResource, metav1.ConditionTrue)
 				})
-				By("Check KV", func() {
+				By("Check SSP", func() {
```
Comment (nunnatsa) [inline pkg/controller/hyperconverged/hyperconverged_controller_test.go:529]: What is the reason for this change?

**Review by nunnatsa — CHANGES_REQUESTED** [review]

File: pkg/controller/hyperconverged/hyperconverged_controller_test.go (hunk 4 above)
Comment (ksimon1) [reply pkg/controller/hyperconverged/hyperconverged_controller_test.go:529]: because the test is checking ssp not kubevirt

File: pkg/controller/hyperconverged/hyperconverged_controller_test.go (hunk 3 above)
Comment (ksimon1) [reply pkg/controller/hyperconverged/hyperconverged_controller_test.go]: ah I see, I updated the test

**Review by nunnatsa — APPROVED** [review]
/approve

**Comment by nunnatsa** [comment]
Adding a new parameter to the API reduces the coverage of the auto-generated code. 
/override coverage/coveralls
